python/gtsam/preamble/nonlinear.h

```diff
@@ -9,4 +9,7 @@
  * automatic STL binding, such that the raw objects can be accessed in Python.
  * Without this they will be automatically converted to a Python object, and all
  * mutations on Python side will not be reflected on C++.
- */
\ No newline at end of file
+ */
+#include <pybind11/stl.h>
```

Comment:
Is this needed? I ran the code in this PR locally and it works without updating this file.

Reply:
Yup just get rid of all the changes to this file and you should be golden. This is why unit testing is important. :slightly_smiling_face:

Reply:
Interesting, it doesn't work for me without it... it fails [here](https://github.com/borglab/gtsam/pull/1239/files#diff-757771908deb6caea6b6e19556f82810b4cf28d344426be172f961f8c266f214R132). I am running Linux 20.04, though, which I just noticed is not one of the supported platforms. Maybe that's the difference?

I'm happy to remove this from the PR if it works on all the currently supported platforms.

Reply:
Do you mean Ubuntu 20.04? I am using the same system and ran it locally to great effect, so it might be some other setting? What compiler are you using? GCC-7?

Reply:
Yeah but with gcc-9 (9.4.0 to be specific)